### ISSUE TITLE
Final Code

### ISSUE THREAD

**Issue by guillaume-guerdoux** [wondervictor/WESPE.pytorch#2]
Hello,
I'm currently trying to implement WESPE with tensorflow but I have some difficulties. I read your code and have a few questions : 
- First, did you finish it ? Is your code working to enhance images ?
- As I read, it seems you don't use residual connections in the generator , why ? (i've quite never used pytorch so I may misread) 
- When you optimize your model, why did you first call   g_optimizer.step() f_optimizer.step() instead of c_optimizer.step() and t_optimizer.step() (because you first want to train your discriminator with texture loss and color loss)
- The training set is composed of original images and not patches ? Because they didn't mention it in the paper 
Thanks :)

**Comment by wondervictor** [comment]
For the lack of enough computation resources, I've just tested my code on the sample set provided by ETH Zurich (DPED Project)
Thanks for these questions, I found that my latest version hadn't been pushed :( .

Residual Connections:

````
class ConvBlock(nn.Module):

    def __init__(self):
        super(ConvBlock, self).__init__()
        self.conv1 = nn.Conv2d(64, 64, 3, padding=1)
        self.conv2 = nn.Conv2d(64, 64, 3, padding=1)
        self.bn = nn.BatchNorm2d(64)

    def forward(self, x):
        
        y = F.relu(self.conv1(x))
        y = self.bn(y)
        y = F.relu(self.conv2(y))
        y = self.bn(y)
        y += x

        return y
````


Optimize Model:

````

# Optimize discriminator:
 c_optimizer.zero_grad()
 t_optimizer.zero_grad()
 color_loss.backward()
 texture_loss.backward()
 c_optimizer.step()
 t_optimizer.step()


# Generator:

 g_optimizer.zero_grad()
 f_optimizer.zero_grad()

 total_loss.backward()

 g_optimizer.step()
 f_optimizer.step()
````

**Comment by RoyiAvital** [comment]
Could you push your latest?
